File: backend/app/model/customer.rb
class Customer
  attr_reader :name, :email, :membership, :orders, :address

  def initialize(name, email, membership = nil, address = nil)
  	@name = name
  	@email = email
  	@orders = []
  	@membership = membership
  	@address = address
  end

  def email
  	@email
  end

  def email=(email)
  	@email = email
  end

  def membership
  	@membership
  end

  def membership=(membership)
  	@membership = membership
  end

  def address
  	@address
  end

  def address=(address)
  	@address = address
  end

  def name
  	@name
  end

  def name=(name)
  	@name = name
  end

end

File: backend/src/main/kotlin/challange/domain/Customer.kt
package main.challenge.domain

import java.time.LocalDateTime

data class Address(val value: String)

data class Discount(val value: Double)

data class Membership(val date: LocalDateTime, val name: String)

class Customer(val name: String, val email: String) {
    private val memberships = mutableListOf<Membership>()

    val allMemberships
        get() = memberships

    fun addMembership(product: Product) {
        this.memberships.add(Membership(LocalDateTime.now(), product.name))
    }
}

